client/signup/steps/subscribing-email/index.jsx

```diff
@@ -1,67 +1,74 @@
-// import config from '@automattic/calypso-config';
-import { useEffect, useState } from '@wordpress/element';
+import { useEffect } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
 import emailValidator from 'email-validator';
 import { localize } from 'i18n-calypso';
 import { connect } from 'react-redux';
-// import { getLocaleSlug } from 'calypso/lib/i18n-utils';
+import useCreateNewAccountMutation from 'calypso/signup/hooks/use-create-new-account';
 import StepWrapper from 'calypso/signup/step-wrapper';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { submitSignupStep } from 'calypso/state/signup/progress/actions';
 import SubscribingEmailStepContent from './content';
 
-// const createNewAccount = async ( props, setIsLoading, setDisplayUserExistsError ) => {
-// 	const email = props.queryParams.email;
-// 	try {
-// 		// eslint-disable-next-line no-undef
-// 		await wpcom.req.post( '/users/new', {
-// 			email: typeof email === 'string' ? email.trim() : '',
-// 			is_passwordless: true,
-// 			signup_flow_name: props.flowName,
-// 			validate: false,
-// 			locale: getLocaleSlug(),
-// 			client_id: config( 'wpcom_signup_id' ),
-// 			client_secret: config( 'wpcom_signup_key' ),
-// 			anon_id: getTracksAnonymousUserId(),
-// 		} );
-// 		// Do stuff with response and redirect
-// 		props.recordTracksEvent( 'calypso_signup_reader_landing_cta' );
-// 		props.submitSignupStep( { stepName: props.stepName } );
-// 		props.goToNextStep();
-// 	} catch ( error ) {
-// 		if ( ! [ 'already_taken', 'already_active', 'email_exists' ].includes( error.error ) ) {
-// 			// Subscribe existing user to guides emails through API endpoint
-// 		}
-// 		setIsLoading( false );
-// 	}
-// };
+function isHttpsOrHttp( url ) {
+	return ( !! url && url.startsWith( 'https://' ) ) || url.startsWith( 'http://' );
+}
 
 function SubscribingEmailStep( props ) {
-	const handleButtonClick = async () => {};
+	const { flowName, goToNextStep, queryParams, stepName } = props;
+	const {
+		mutate: createNewAccount,
+		isError,
+		isSuccess,
+		isPending,
+		error,
+	} = useCreateNewAccountMutation();
 
-	const { flowName, positionInFlow, queryParams, stepName } = props;
-	const [ isLoading, setIsLoading ] = useState( true );
+	const redirectUrl = isHttpsOrHttp( queryParams?.redirect_to )
+		? addQueryArgs( queryParams.redirect_to, { subscribed: true } )
+		: addQueryArgs( 'https://' + queryParams.redirect_to, { subscribed: true } );
 
 	useEffect( () => {
-		const email = queryParams.email;
-
-		if ( emailValidator.validate( email ) ) {
-			// TODO: Fetch request to create new account will replace this placeholder timeout
-			setTimeout( () => setIsLoading( false ), 5000 );
+		if ( emailValidator.validate( queryParams.user_email ) ) {
+			createNewAccount( {
+				userData: {
+					email: typeof queryParams.user_email === 'string' ? queryParams.user_email.trim() : '',
```

Comment:
I think it might be a good idea to lift the `typeof queryParams.user_email === 'string' ? ...` check prior to the `validate()` and `createNewAccount()` calls. We would want to avoid calling `createNewAccount` with an empty string for the email address, and I think it makes sense to call `validate()` with the same trimmed value.

So something like:

```
const email = typeof queryParams.user_email === 'string' ? queryParams.user_email.trim() : ''

if ( emailValidator.validate( email ) ) {
  createNewAccount( {
    userData: {
      email,
      ...
```
```

Reply:
Totally agree making the change now

Reply:
[9179d12](https://github.com/Automattic/wp-calypso/pull/92013/commits/9179d12d962c0cd58aabb956012cc961e23a1007)